lib/navigation/view/webpage_header.dart

```diff
@@ -96,9 +96,55 @@ class _ProfileDropdownMenu extends StatelessWidget {
 
   final double headerHeight;
 
+  Map<int, _PopupItem> get _popupItems => {
+        1: _PopupItem(
```

Comment:
You made it a class anyway, so don't half-ass it. Make it a widget and use it to actually get rid of duplicate code here. E.g.
```
_PopupItem(
    onTap: AppNavigator.pushNamed(context, SettingsPage.routeName),
    icon: Icons.settings,
    text: S.current.navigationSettings
)
```

And in the widget's build method you should have something like `return PopupMenuItem(...)`

Reply:
sure, great idea

Reply:
https://github.com/student-hub/acs-upb-mobile/pull/298/commits/524f08a5b631f2e9fb0e3080c94e7ff1c85f55cf solved, still a little bit of duplication, don't know about a map builder to solve that

Reply:
https://github.com/student-hub/acs-upb-mobile/pull/298/commits/e6efe21cb2ed946eaa46907ac009b2d06d955609 this should make it as easy as possible to add items to the menu